workflows/metatdenovo.nf

```diff
@@ -73,17 +144,373 @@ workflow METATDENOVO {
+        ch_versions = ch_versions.mix(EGGNOG.out.versions)
+        ch_merge_tables = EGGNOG.out.sumtable
+    } else {
+        ch_protein
+            .map { meta, protein -> [ meta, [] ] }
+            .set { ch_merge_tables }
+    }
+
+
+    //
+    // SUBWORKFLOW: run kofamscan on the ORF-called amino acid sequences
+    //
+    if( !params.skip_kofamscan ) {
+        ch_protein
+            .map { meta, protein -> [ meta, protein ] }
+            .set { ch_kofamscan }
+        KOFAMSCAN( ch_kofamscan, ch_fcs_for_summary)
+        ch_versions = ch_versions.mix(KOFAMSCAN.out.versions)
+        ch_kofamscan_summary = KOFAMSCAN.out.kofamscan_summary.collect().map { meta, tsv -> tsv }
+        ch_merge_tables
+            .combine( ch_kofamscan_summary )
+            .set { ch_merge_tables }
+    } else {
+        ch_merge_tables
+            .map { meta, tsv -> [ meta, tsv, [] ] }
+            .set { ch_merge_tables }
+    }
+
+    // set up contig channel to use in CAT and TransRate
+    UNPIGZ_CONTIGS(ch_assembly_contigs)
+    ch_unzipped_contigs = UNPIGZ_CONTIGS.out.unzipped
+    ch_versions = ch_versions.mix(UNPIGZ_CONTIGS.out.versions)
+
+    //
+    // MODULE: Use TransRate to judge assembly quality, piped into MultiQC
+    //
+    TRANSRATE(ch_unzipped_contigs)
+    ch_versions = ch_versions.mix(TRANSRATE.out.versions)
+
+    //
+    // SUBWORKFLOW: Eukulele
+    //
+    ch_eukulele_db = Channel.empty()
+    if( !params.skip_eukulele){
+        // Create a channel for EUKulele either with a named database or not. The latter means a user-provided database in a directory.
+        if ( params.eukulele_db ) {
+            Channel
+                .of ( params.eukulele_db.split(',') )
+                .map { [ it, file(params.eukulele_dbpath) ] }
+                .set { ch_eukulele_db }
+        } else {
+            Channel.fromPath(params.eukulele_dbpath, checkIfExists: true)
+                .map { [ [], it ] }
+                .set { ch_eukulele_db }
+        }
+        ch_protein
+            .map { meta, protein -> [ [ id:"${meta.id}" ], protein ] }
+            .combine( ch_eukulele_db )
+            .set { ch_eukulele }
+        SUB_EUKULELE( ch_eukulele, ch_fcs_for_summary )
+        ch_taxonomy_summary = SUB_EUKULELE.out.taxonomy_summary.collect().map { meta, tsv -> tsv }
+        ch_versions = ch_versions.mix(SUB_EUKULELE.out.versions)
+        ch_merge_tables
+            .combine( ch_taxonomy_summary )
+            .set { ch_merge_tables }
+    } else {
+        ch_merge_tables
+            .map { meta, tsv1, tsv2 -> [ meta, tsv1, tsv2, [] ] }
+            .set { ch_merge_tables }
+    }
+
+    //
+    // MODULE: Collect statistics from mapping analysis
+    //
+    if( !params.skip_eggnog  || !params.skip_eukulele || !params.skip_kofamscan) {
+        MERGE_TABLES ( ch_merge_tables )
+        ch_collect_stats
+            .combine(MERGE_TABLES.out.merged_table.collect{ meta, tblout -> tblout }.map { [ it ] })
```

Comment:
Another list of a list instance `map { [ it ] })`.

Reply:
see comments above